Sources/IBMSwiftSDKCore/Authentication/IAMAuthenticator.swift

```diff
@@ -70,28 +73,27 @@ class IAMTokenSource: TokenSource {
     }
 
     func getToken(completionHandler: @escaping (String?, RestError?) -> Void) {
-        // request a new access token if the current token is expired
-        guard let refreshDate = refreshDate, refreshDate.timeIntervalSinceNow > 0 else {
-            requestToken {
-                token, error in
-
-                guard let token = token, error == nil else {
-                    completionHandler(nil, error)
-                    return
-                }
-                self.token = token.accessToken
-                // Compute refreshDate for token
-                let expirationDate = Date(timeIntervalSince1970: Double(token.expiration))
-                // We want a little buffer to make sure we refresh proactively
-                let buffer = expirationDate.timeIntervalSinceNow * -0.2
-                self.refreshDate = expirationDate.addingTimeInterval(buffer)
-                completionHandler(self.token, nil)
+        // If we have a token, pass it to the completion handler.
+        if let token = self.token {
+            completionHandler(token, nil)
+            if let refreshDate = refreshDate, refreshDate.timeIntervalSinceNow < 0 {
+                refreshQueue.async{ self.refreshToken() }
             }
+        } else {
+            requestToken(completionHandler: completionHandler)
+        }
+    }
+
+    // request a new access token if he refresh time for the current token is passed
```

Comment:
nit: `he` -> `the`

Reply:
I must have fixed this since I can't find it any longer.